Makefile

```diff
@@ -84,6 +84,13 @@ CONTAINER_TOOL ?= podman
 SHELL = /usr/bin/env bash -o pipefail
 .SHELLFLAGS = -ec
 
+# Fetches the current OS & arch details to be used during image builds.
+# export PLATFORM to required os/arch to build images for preferred platform.
+# for ex; PLATFORM=linux/amd64 make docker-build
+CURRENT_OS ?= $(shell uname -s | tr A-Z a-z)
+CURRENT_ARCH ?= $(shell uname -m | tr A-Z a-z)
+PLATFORM ?= $(CURRENT_OS)/$(CURRENT_ARCH)
```

Comment:
@ayushsatyam146 @sayan-biswas what happens if you run this on Apple Silicon machines?

IMO we should force the OS to be `linux` no matter what - we don't support deploying the operator on Windows nodes. I also don't think there is a container runtime that is native to MacOS (`darwin`).